client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import omitBy from 'lodash/omitBy';
+import pick from 'lodash/pick';
+import QueryString from 'querystring';
+import tinymce from 'tinymce/tinymce';
+
+/**
+ * Internal dependencies
+ */
+import shortcodeUtils from 'lib/shortcode';
+import { getVideo } from 'state/videos/selectors';
+import QueryVideo from 'components/data/query-video';
+
+class WpVideoView extends Component {
+
+	static match( content ) {
+		const match = shortcodeUtils.next( 'wpvideo', content );
+
+		if ( match ) {
+			return {
+				index: match.index,
+				content: match.content,
+				options: {
+					shortcode: match.shortcode
+				}
+			};
+		}
+	}
+
+	static serialize( content ) {
+		return encodeURIComponent( content );
+	}
+
+	constrainVideoDimensions( shortcodeWidthAttribute, shortcodeHeightAttribute ) {
+		const defaultWidth = 640;
+		const defaultAspectRatio = 16 / 9;
+		let { width, height } = this.props.video
+			? this.props.video
+			: {
+				width: defaultWidth,
+				height: defaultWidth / defaultAspectRatio
+			};
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			const definedAspectRatio = shortcodeWidthAttribute / shortcodeHeightAttribute;
+			if ( definedAspectRatio > aspectRatio ) {
+				width = shortcodeHeightAttribute * aspectRatio;
+				height = shortcodeHeightAttribute;
+			} else {
+				width = shortcodeWidthAttribute;
+				height = shortcodeWidthAttribute / aspectRatio;
+			}
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions(
+			parseInt( namedAttrs.w, 10 ),
+			parseInt( namedAttrs.h, 10 )
+		);
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: namedAttrs.autoplay === 'true',
```

Comment:
>I found no official documentation w.r.t `0`/`1` and `true`/`false` usage for boolean shortcode attributes, but both flavors seem to used in the wild.

Yeah, unfortunately this is one of those things which aren't very well standardized and left to the plugin author to choose.

Your changes look good here.